### ISSUE TITLE
nixFlakes: `nix copy` tries to build derivations it should just copy

### ISSUE THREAD

**Issue by colemickens** [NixOS/nix#3766]
**Describe the bug**

I have a `nix copy` command that just copies a `drv` and dependents to a `file:///tmp/path` store. It works on stable nix, but not with `nixFlakes`, as it complains about aarchitecture issues.

What I'm doing is instantiating my "rpi4" system configuration, then copying it to a dir, then rsyncing it to the builder. AFAIK this should work.

```

# <investigate>
# looks like: regression when copying aarch64 derivs from x86->aarch64!!!
# - the first command fails
# - the second one succeeds, doesn't seem right
# first error:
  # + nix --experimental-features 'nix-command flakes' --pure-eval --show-trace copy --to file:///tmp/colemickens-nixtmp /nix/store/7dmhqf7a0w0y128gz1xv3wy55ankcrwm-nixos-system-raspberry-20.09.git.drv
  # error: --- Error ---------------------------------------------------------------------------------------- nix
  # a 'aarch64-linux' with features {} is required to build '/nix/store/9psjdx08hy7wbm3hap4lcsgagam8p660-hm_.configzsh.zshrc.drv', but I am a 'x86_64-linux' with features {benchmark, big-parallel, kvm, nixos-test}
nix copy --to "file://${t}" "${drv}"
nix-shell -I nixpkgs=https://github.com/nixos/nixpkgs/archive/nixos-unstable.tar.gz -p nix --command "nix copy --to file://${t} ${drv}"
# </investigate>
```

error: `  # a 'aarch64-linux' with features {} is required to build '/nix/store/9psjdx08hy7wbm3hap4lcsgagam8p660-hm_.configzsh.zshrc.drv', but I am a 'x86_64-linux' with features {benchmark, big-parallel, kvm, nixos-test}`

But it really shouldn't be trying to build that AFAIK.

**Steps To Reproduce**

1. `nix-instantiate` a derivation for an `aarch64-linux` system on an `x86_64-linux`.
2. Attempt to `nix copy --to file:///tmp/store ${drv}`.

**Expected behavior**

It should work.

Good nix version: `nix (Nix) 2.4pre20200622_334e26b
`

Bad nix version: `2.3.6`

**Additional context**

**Comment by dasJ** [comment]
Duplicate of #3696 which is fixed by #3762

**Comment by colemickens** [comment]
Thanks @dasJ, I'll follow those.
